src/db/mysql/batch.rs

```diff
@@ -0,0 +1,272 @@
+#![allow(proc_macro_derive_resolution_fallback)]
+
+use diesel::{
+    self, dsl::sql, insert_into, sql_types::Integer, update, ExpressionMethods, OptionalExtension,
+    QueryDsl, RunQueryDsl, TextExpressionMethods,
+};
+use serde_json;
+
+use super::{
+    models::{MysqlDb, Result},
+    schema::batches,
+};
+use db::{params, results, DbError};
+
+/// Rough guesstimate of the maximum reasonable life span of a batch.
+pub const BATCH_LIFETIME: i64 = 2 * 60 * 60 * 1000; // 2 hours, in milliseconds
+
+pub fn create(db: &MysqlDb, params: params::CreateBatch) -> Result<results::CreateBatch> {
+    let user_id = params.user_id.legacy_id as i32;
+    let collection_id = db.get_collection_id(&params.collection)?;
+    let timestamp = db.timestamp().as_i64();
+    // Sync timestamps are in seconds and quantized to two decimal places, so
+    // when we convert one to a bigint in milliseconds, the final digit is
+    // always zero. But we want to use the lower digits of the batchid for
+    // sharding writes via (batchid % num_tables), and leaving it as zero would
+    // skew the sharding distribution.
```

Comment:
Do you in fact do this sharding across multiple "batches" tables?

Reply:
Nope, so I removed this comment and the associated code